clang/include/clang/AST/TypeLoc.h

```diff
@@ -243,7 +243,7 @@ class TypeLoc {
 
 inline TypeSourceInfo::TypeSourceInfo(QualType ty, size_t DataSize) : Ty(ty) {
   // Init data attached to the object. See getTypeLoc.
-  memset(this + 1, 0, DataSize);
+  memset(const_cast<void *>(static_cast<const void *>(this + 1)), 0, DataSize);
```

Comment:
```suggestion
  memset(static_cast<void *>(this + 1), 0, DataSize);
```

This should be enough, right?

Reply:
My intention here was to match the double cast in `::getTypeLoc()`, so the way this memory is referenced is the same throughout the impls.

I've just tested the single cast, and it does not trigger the warning as well. Will amend my patch then.

Reply:
> I've just tested the single cast, and it does not trigger the warning as well. Will amend my patch then.

Done just now.